packages/nimble-components/src/menu-button/testing/menu-button.pageobject.ts

```diff
@@ -0,0 +1,84 @@
+import {
+    keyArrowDown,
+    keyArrowUp,
+    keyEnter,
+    keyEscape,
+    keySpace
+} from '@microsoft/fast-web-utilities';
+import type { MenuButton } from '..';
+import { createEventListener } from '../../utilities/tests/component';
+
+/**
+ * Page object for `nimble-menu-button` component to provide consistent ways
+ * of query and interacting with the component during tests.
+ */
+export class MenuButtonPageObject {
+    public constructor(protected readonly menuButtonElement: MenuButton) {}
+
+    public isOpen(): boolean {
+        return this.menuButtonElement.open;
+    }
+
+    public clickMenuButton(): void {
+        // Focus the menu button before calling click() because ordinarily a mouse click
+        // would bring focus to the button, but calling click() directly does not.
+        this.menuButtonElement.focus();
+        this.menuButtonElement.toggleButton!.control.click();
+    }
+
+    public async openMenu(): Promise<void> {
+        if (this.isOpen()) {
+            return;
+        }
+
+        const toggleListener = createEventListener(
+            this.menuButtonElement,
+            'toggle'
+        );
+        this.clickMenuButton();
+        await toggleListener.promise;
+    }
+
+    public pressEnter(): void {
```

Comment:
It probably wasn't clear, so I'll update the docs comment for the page object ... the page object is really intended to be for interacting with the menu button itself (not the menu items). I guess the one exception is the function for pressing ESC on the menu, though technically, that's not a key press on the menu itself.

Reply:
Yeah, I would expect a menu button page object to encompass all the interactions for the menu button, meaning both the button and the menu. Otherwise, wouldn't it just be a toggle button page object?

Reply:
I think even without menu interactions, this is more than just a toggle button page object. Specifically, this includes:
- all the keypress/keydown code that needs to be called on the right part of the shadowroot
- having an async function that knows how to wait for the menu to fully open before returning

Perhaps if we had a page object for the `nimble-menu`, `nimble-menu-item`, and/or `nimble-anchor-menu-item`, we could expose a reference to those page objects here. However, I think a key thing to keep in mind is that the menu and its items are being slotted into the menu button by the client app. As a result, a client test can interact with those items directly in tests without having to make any assumptions about the internal workings or structure of the menu button.

Reply:
Good points!